proposed/2021/DotNetAudit.md

```diff
@@ -0,0 +1,296 @@
+
+Found 1 top-level Legacy deprecated package(s) in 36 scanned packages.
+
+Run 'dotnet audit fix' to fix them.
+
+warning: Outdated packages found!
+[net5.0]:
+Top-level Package						            Resolved	Latest
+> anurse.testing.TestDeprecatedPackage	            1.0.0		2.0.0
+> UmbracoForms							            8.4.1	    8.7.1
+
+Found 2 top-level outdated package(s) in 36 scanned packages.
+
+Run 'dotnet audit fix' to fix them.
+```
+
+##### Endpoints
+
+NuGet will use existing endpoints to optimize the speed of audit results.
+
+- [Deprecation](https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#package-deprecation)
+- [Vulnerability](https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#vulnerabilities)
+- Outdated - No existing endpoint, will need to call a source.
+
+##### dotnet audit --json
+
+To get a detailed audit report in a JSON format.
+
+```
+{
+  "audit": {
+    "vulnerabilities": {
+      "low": "0",
+      "moderate": "2",
+      "high": "0",
+      "critical": "0",
+      "dependencies": [
+        {
+          "name": "UmbracoForms",
+          "requestedVersion": "8.4.1",
+          "resolvedVersion": "8.4.1",
+          "vulnerabilitySeverity": "Moderate",
+          "advisoryUrl": "https://github.com/advisories/GHSA-8m73-w2r2-6xxj",
+          "transitiveDependencies": [
+            {
+              "name": "Microsoft.Data.OData",
+              "resolvedVersion": "5.2.0",
+              "severity": "Moderate",
+              "advisoryUrl": "https://github.com/advisories/GHSA-mv2r-q4g5-j8q5"
+            }
+          ]
+        }
+      ]
+    },
+    "deprecations": {
+      "legacy": "1",
+      "critical-bugs": "0",
+      "other": "0",
+      "dependencies": [
+        {
+          "name": "anurse.testing.TestDeprecatedPackage",
+          "requestedVersion": "1.0.0",
+          "resolvedVersion": "1.0.0",
+          "reason": "Legacy",
+          "alternativeDependencies": [
+            {
+              "name": "Microsoft.AspNetCore.MVC",
+              "version": "0.0.0"
+            }
+          ]
+        }
+      ]
+    },
+    "outdated": {
+      "packages": "2",
+      "dependencies": [
+        {
+          "name": "anurse.testing.TestDeprecatedPackage",
+          "resolvedVersion": "1.0.0",
+          "latestVersion": "2.0.0"
+        },
+        {
+          "name": "UmbracoForms",
+          "resolvedVersion": "8.4.1",
+          "latestVersion": "8.7.1"
+        }
+      ]
+    },
+    "scannedDependencies": "36"
+  }
+}
+```
+
+##### dotnet audit Exit Codes
+
+- 0 - The command will exit with a 0 exit code if no vulnerabilities, deprecations, or outdated packages were found.
+- 1 - The command will exit with a 1 exit code if a vulnerability, deprecation, or outdated package was found.
+- 2 - The command will exit with a 2 exit code if it unexpectedly failed.
+- 3 - The command will exit with a 3 exit code if an unsupported project is detected.
```

Comment:
What's the value of splitting exit codes 2 and 3? Would it be simpler to have an exit code for "NuGet was unable to determine if there vulnerabilities, deprecations, or oudated packages. This may due to unexpected failure or unsupported projects"

Reply:
Personally I think unsupported vs. failure are different scenarios. Why have someone guess?